spring-cloud-gcp-docs/src/main/asciidoc/datastore.adoc

```diff
@@ -0,0 +1,404 @@
+}
+----
+
+This setup takes care of bringing in the latest compatible version of Cloud Java Cloud Datastore libraries
+as well.
+
+A https://github.com/spring-cloud/spring-cloud-gcp/tree/master/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample[sample application]
+is available.
+
+=== Configuration
+
+To setup Spring Data Cloud Datastore, you have to configure the following:
+
+* Setup the connection details to Google Cloud Datastore.
+
+==== Cloud Datastore settings
+
+You can the use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-datastore[Spring Boot
+Starter for Spring Data Datastore] to autoconfigure Google Cloud Datastore in your Spring application.
+It contains all the necessary setup that makes it easy to authenticate with your Google Cloud
+project.
+The following configuration options are available:
+
+|===
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.datastore.project-id` | GCP project ID where the Google Cloud Datastore API
+is hosted, if different from the one in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>>
+| No |
+| `spring.cloud.gcp.datastore.credentials.location` | OAuth2 credentials for authenticating with the
+Google Cloud Datastore API, if different from the ones in the
+<<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> | No |
+| `spring.cloud.gcp.datastore.credentials.encoded-key` | Base64-encoded OAuth2 credentials for authenticating with the
+Google Cloud Datastore API, if different from the ones in the
+<<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> | No |
+| `spring.cloud.gcp.datastore.credentials.scopes` |
+https://developers.google.com/identity/protocols/googlescopes[OAuth2 scope] for Spring Cloud GCP
+Cloud Datastore credentials | No | https://www.googleapis.com/auth/datastore
+| `spring.cloud.gcp.datastore.namespace` | The Cloud Datastore namespace to use | No | the Default namespace of Cloud Datastore in your GCP project
+|===
+
+==== Repository settings
+
+Spring Data Repositories can be configured via the `@EnableDatastoreRepositories` annotation on your
+main `@Configuration` class.
+With our Spring Boot Starter for Spring Data Cloud Datastore, `@EnableDatastoreRepositories` is automatically
+added.
+It is not required to add it to any other class, unless there is a need to override finer
+grain configuration parameters provided by
+https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java[`@EnableDatastoreRepositories`].
+
+==== Autoconfiguration
+
+Our Spring Boot autoconfiguration creates the following beans available in the Spring application context:
+
+- an instance of `DatastoreTemplate`
+- an instance of all user defined repositories extending `CrudRepository` or `PagingAndSortingRepository`, when repositories are enabled
```

Comment:
When using the autoconfig it's enabled for them. However, if they are doing their own config then they can choose to not use this annotation and they wouldn't be able to use the repo features.

I don't think we would want the user to enable the use of this module with `spring.cloud.gcp.datastore.enabled=true` but then to also disable to use of the repositories, since that is the main feature we are providing.